policy/pa.go

```diff
@@ -285,3 +286,28 @@ func (pa *AuthorityImpl) ChallengesFor(identifier core.AcmeIdentifier) ([]core.C
 
 	return shuffled, shuffledCombos
 }
+
+// ExtractDomainIANASuffix returns the public suffix of the domain using only the "ICANN"
+// section of the Public Suffix List database.
+// If the domain does not end in a suffix that belongs to an IANA-assigned
+// domain, ExtractDomainIANASuffix returns an error.
+func extractDomainIANASuffix(name string) (string, error) {
+	if name == "" {
+		return "", fmt.Errorf("Blank name argument passed to ExtractDomainIANASuffix")
```

Comment:
Is there a problem with passing through empty strings to `Find()`? I assume it will return an appropriate error? If we choose to keep this check here, it should get a test case.